santoku/utils/url_handler.py

```
@@ -0,0 +1,326 @@
+            except for URLs containing IP addresses. Set to `False` by default.
+
+        Returns
+        -------
+        str
+            The partial domain of the `url`. If `return_invalid_urls` set to `False`, empty string
+            will be returned for invalid urls, otherwise, if subdomain is detected, num_subdomains +
+            domain will be returend, if no subdomain is detected, the URL will be returned as it is.
+
+        Notes
+        -----
+        Notice that with our definition of invalid URLs, URLs containing IP addresses will be
+        considered as invalid as they do not contain suffixes.
+
+        Examples
+        --------
+        https://sub2.sub1.example.com.es/path, 0 -> example.com.es
+        https://sub2.sub1.example.com.es/path, 1 -> sub1.example.com.es
+        https://sub2.sub1.example.com.es/path, 2 -> sub2.sub1.example.com.es
+        https://sub2.sub1.example.com.es/path, 3 -> sub2.sub1.example.com.es
+
+        """
+        res = tldextract.extract(url)
+
+        # When URL is invalid
+        if not res.domain or not res.suffix:
+
+            # If URL contain a domain or a suffix, we remove the scheme and path
+            if return_invalid_urls:
+
+                if res.domain:
+                    if res.subdomain and num_subdomains > 0:
+                        # URL contain subdomain and domain, we return the last n subdomains + domain
+                        component = res.subdomain.split(".")[-num_subdomains:] + [res.domain]
+                        return ".".join(component)
+
+                    # URL only contain domain, we return only the domain
+                    return res.domain
+
+                elif res.suffix:
+                    # URL contain only sufix, we return only the suffix
+                    return res.suffix
+
+                # If URL doesn't contain anything identified as domain or suffix, return as it is
+                return url
+
+            return ""
```

Shall we raise an exception?